src/hash.ts

```diff
@@ -14,6 +14,17 @@ export function djb2(str: string) {
         hash = ((hash << 5) + hash) ^ c;
     }
 
-    // JS trickery to convert to unsigned 32 bits
+    // JS trickery to convert to unsigned 32 bits (truncates)
     return hash >>> 0;
 }
+
+/**
+ * Return the hash of `str` but scaled to fit in [0, `window`].
+ */
+export function hashInWindow(str: string, window: number): number {
+    let hash = djb2(str);
+    hash /= 4_294_967_295; // unsigned max
+    hash *= window; // fit window
+
+    return Math.ceil(hash);
```

Comment:
This `Math.ceil` is what was missing from the code. The refactor to break out a separate function was for easier testing.